src/evergreen/api.py

```diff
@@ -949,15 +950,23 @@ def num_of_tests_by_task(self, task_id: str) -> int:
         url = self._create_url(f"/tasks/{task_id}/tests/count")
         return int(self._call_api(url).text)
 
-    def manifest_for_task(self, task_id: str) -> Manifest:
+    def manifest_for_task(self, task_id: str) -> Optional[Manifest]:
         """
         Get the manifest for the given task.
 
         :param task_id: Task Id fo query.
         :return: Manifest for the given task.
         """
         url = self._create_url(f"/tasks/{task_id}/manifest")
-        return Manifest(self._call_api(url).json(), self)  # type: ignore[arg-type]
+
+        manifest: Optional[Manifest] = None
+        try:
```

Comment:
Are we going to modify this logic once [EVG-19971](https://jira.mongodb.org/browse/EVG-19971) ticket is implemented?

Reply:
We could create a DAG ticket that depends on EVG-19971 to come back to it. It's fine for now, seems no better way to do it, but obviously it's preferably not to rely on error message text.

Reply:
Im ok with creating a DAG ticket that depends on EVG-19971